### PR TITLE
Boost: Disable Page Cache on Atomic

### LINKED ISSUE #35856: [Boost] Disable Cache feature on Atomic
Atomic already has caching, so we can't use Page Caching there.

p1708577197947199-slack-C016BBAFHHS

### PR DESCRIPTION
**Pull request by dilirity**
Fixes #35856

## Proposed changes:

* Disable Page Cache feature toggle;
* Add a notice explaining why feature is disabled.

CleanShot 2024-02-22 at 16 04 01

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

n/a

## Does this pull request change what data or activity we track or use?

no

## Testing instructions:

* Setup on Atomic and make sure the feature can't be enabled and notice is properly visible.

### CODE REVIEW

File: projects/plugins/boost/app/assets/src/js/pages/index/index.tsx [hunk 1]
```diff
@@ -127,12 +127,23 @@ const Index = () => {
...
+								'jetpack-boost'
+							) }
+						</p>
+						{ site.isAtomic && (
+							<Notice
+								level="warning"
+								title={ __( 'Page Cache is unavailable', 'jetpack-boost' ) }
+								hideCloseButton={ true }
+							>
+								<p>{ __( 'Atomic sites already have caching enabled.', 'jetpack-boost' ) }</p>
```
Comment (donnchawp) [inline projects/plugins/boost/app/assets/src/js/pages/index/index.tsx]: Looks great, but I would change the warning message:
```suggestion
								<p>{ __( 'Your website already has a page cache running on it powered by WordPress.com.', 'jetpack-boost' ) }</p>
```

**Review by donnchawp — CHANGES_REQUESTED** [review]
Nice! Good to see the greyed out toggle button and the warning. I would change the text of the warning though as the user may not know what "Atomic" is.

**Review by donnchawp — APPROVED** [review]
Looks great!

**Comment by dilirity** [comment]
Cherry picked in 6c6820f for beta 3.1.0.

**Comment by thingalon** [comment]
Beautiful. THanks @dilirity

**Pull request merged**